### ISSUE TITLE
cpprestsdk needs boost

### ISSUE THREAD

**Issue by madebr** [Microsoft/cpprestsdk#991]
Hello,

I'm using cmake to build my project that uses cpprestsdk.
My CMakeLists.txt uses `find_package(cpprestsdk REQUIRED CONFIG)` to get the `cpprestsdk::cpprest` cmake target.
Linking fails because of a missing symbol: `libboost_system.so`.
Adding `Boost::system` to the `target_link_libraries` solves the problem.

I believe this is because of this:
https://github.com/Microsoft/cpprestsdk/blob/744f33baec14abc07b0cf527f25014a6d0cba7e4/Release/cmake/cpprestsdk-config.in.cmake#L14-L20
The ` AND OFF` must disappear.

**Comment by madebr** [comment]
This happens using the fedora packages on linux using gcc-8.

**Comment by ChrisCates** [comment]
You actually don't need to add all those dependencies to cpprestsdk.
You can view my standard CMakeLists.txt for cpprestsdk here: https://github.com/ChrisCates/YungCPP/blob/master/CMakeLists.txt

**Comment by madebr** [comment]
So cpprestsdk does not work out of the cmake box atm.
Maybe the cpprestsdk cmake config should support COMPONENTS?

**Comment by ChrisCates** [comment]
@madebr, makes sense. @BillyONeal, does this make sense to you? This should probably be refactored.

**Comment by wegylexy** [comment]
With C++17 or C++20 coming soon, can cpprestsdk go more pure without Boost?

**Comment by BillyONeal** [comment]
C++17 and C++20 don't have replacements for anything that lead to the boost dependency.

**Comment by ChrisCates** [comment]
@BillyONeal, it essentially just requires a component or module in CMake.
It would also be nice to prepare module based builds for this project say for the http requests, websocket listener, and, http server/wss server.

But, as I see it. There are a lot of nice-to-haves in the task list and I know you're busy with other things.

**Comment by Clovel** [comment]
It would be nice to have a Boost-free `spprestsdk`. But at the moment it isn't. Hence the need to add back the Boost CMake dependency. 

When in the future this project goes Boost-free, we will delete the dependency. 

Until then, we can't use the CMake package mechanism as intended.

**Comment by Clovel** [comment]
`COMPONENTS` are a good idea. If you need examples, a very effective example is the Qt project, which uses components extensively and work very well with CMake.
